docs/tutorial/graph_api.rst

```diff
@@ -0,0 +1,598 @@
+operations, similar to CPU threads. Adding operation A before operation B to a stream ensures A happens before B,
+regardless of any interdependencies (or lack thereof) between them. A stream can be thought of as a first-in, first-out
+(FIFO) queue of operations.
+
+Multiple streams operate independently, and manual synchronization is required when dependencies cross stream
+boundaries. Additionally, each operation in a stream is scheduled independently, which — depending on the complexity of
+the enqueued operation — might lead to noticeable CPU launch overhead and kernel dispatch latency, especially for
+workloads with many small kernels. However, applications that use streams are well suited for workloads that are
+dynamic and unpredictable.
+
+For more information about HIP streams, see :ref:`asynchronous_how-to`.
+
+Graphs
+^^^^^^
+
+HIP graphs model dependencies between operations the way developers naturally think about them — as nodes and edges on
+a diagram. Each node in the graph represents an operation, and each edge represents a dependency between two nodes. If
+no edge exists between two nodes, they are independent and can execute in any order.
+
+Because dependency information is built into the graph, the HIP runtime automatically inserts the necessary
+synchronization points. Launching all operations in a graph requires only a single API call, reducing launch overhead
+and dispatch latency to near-zero. This is especially beneficial for workloads with many small kernels, where launch
+overhead can dominate overall execution time.
+
+Graphs must be defined once before use, making them ideal for fixed workflows that run repeatedly. While node
+parameters can be updated between executions, the graph structure itself cannot change after instantiation. This
+structural immutability is the primary trade-off compared to the flexibility of streams.
+
+For more information about HIP graphs, see :ref:`how_to_HIP_graph`.
+
+When to use graphs
+^^^^^^^^^^^^^^^^^^
+
+.. list-table::
+  :header-rows: 1
+  :class: decision-matrix
+
+  * - ✅ **Use Graphs When**
+    - ❌ **Avoid Graphs When**
+  * - Workflow is fixed and repetitive
+    - Workflow changes dynamically
+  * - Same kernels execute many times
+    - One-shot operations
+  * - Launch overhead is significant (many small kernels)
+    - Kernels are long-running
+
+Transitioning a CT reconstruction pipeline
+------------------------------------------
+
+In this tutorial, we will modify an existing GPU-accelerated stream-based image processing pipeline that reconstructs
+computer tomography (CT) data (the classic Shepp-Logan phantom [ShLo74]_). The pipeline transforms raw X-ray
+projections into clear cross-sectional images used in medical diagnosis.
+
+.. figure:: ../data/tutorial/graph_api/ct_reconstruction_overview.png
+  :alt: Diagram showing raw projection data being transformed into a reconstructed CT slice
+  :align: center
+
+The reconstruction pipeline consists of:
```

Comment:
You can mention in front of this the data generation. Those will appear on the trace files, developers should know, that data generation also multi streamed, but not part of the graph lunch focus.